app/controllers/account_requests_controller.rb

```diff
@@ -0,0 +1,56 @@
+class AccountRequestsController < ApplicationController
+  skip_before_action :authorize_user
+  skip_before_action :authenticate_user!
+
+  before_action :set_account_request_from_token, only: [:received, :confirmation, :confirm]
+
+  layout 'devise'
+
+  def received; end
+
+  def confirmation; end
+
+  def confirm
+    @account_request.update!(confirmed_at: Time.current)
+    AccountRequestMailer.approval_request(account_request_id: @account_request.id).deliver_later
+  end
+
+  def invalid_token; end
+
+  def new
+    @account_request = AccountRequest.new
+  end
+
+  def create
+    @account_request = AccountRequest.new(account_request_params)
+
+    if @account_request.save
+      AccountRequestMailer.confirmation(account_request_id: @account_request.id).deliver_later
+
+      redirect_to received_account_requests_path(token: @account_request.identity_token),
+                  notice: 'Account request was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_account_request_from_token
+    @account_request = AccountRequest.get_by_identity_token(params[:token])
```

Comment:
Ok, so I'm actually writing this comment last even though it's appearing near the top:

Would it be a bad idea to treat the JWT as the `:id` and then treat the actions that use this for lookup as member routes? (cf. my comment on `config/routes.rb` below)

Reply:
I think it might lead to some extra confusion to treat the token as an id. The tokens go through a different logic pathway and I wanted to make sure they stay separate. I wouldn't want people to assume the token is the primary key.